File: misc/calgary/keeping_time.py
n,a,b = [int(i) for i in input().split()]

best = -1

t = 0
go = True
while t % b != 0 or go:
    go = False

    if (t+n) % b == 0:
        best = t+n
        break

    t += a

t = 0
go = True
while t % a != 0 or go:
    go = False

    if (t+n) % a == 0:
        best = min(best, t+n)
        if best == -1:
            best = t+n
        break

    t += b

print(best)

File: misc/calgary/keeping_time.cpp
#include <iostream>
using namespace std;

int main(){
	long long n,a,b;
	cin>>n>>a>>b;

	long long best=-1;

	long long t=0;
	do{
		if((t+n)%b==0){
			best=t+n;
			break;
		}
		t+=a;
	}while(t%b!=0);

	t=0;
	do{
		if((t+n)%a==0){
			if(t+n<best||best<0)
				best=t+n;
			break;
		}
		t+=b;
	}while(t%a!=0);

	cout<<best<<'\n';
}
